### PR TITLE
Auth: Enable case insensitive logins/emails by default

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**

Enables lower case sql queries for all users login, emails across grafanas users. 

**Why do we need this feature?**

To make logins and emails lowercased across grafanas users and enforce use of lowercase in signing up for Grafana. This is to consolidate a standard.

# Release notice breaking change
This is a breaking change for users who use uppercase in their login or emails. The users are by default now using lowercase as part of their login and emails.

Before this code change, users would be able to still log in as either `aUser@user.com` or `auser@user.com`, users are now only able to login and signup with grafana using lowercasing `auser@user.com`.

We recommend reviewing our [blog post](https://grafana.com/blog/2022/12/12/guide-to-using-the-new-grafana-cli-user-identity-conflict-tool-in-grafana-9.3/#:~:text=A%20user%20identity%20conflict%20occurs,more%20capitalized%20letters%20%E2%80%9Cgrafana_LOGIN%E2%80%9D.)  about using the CLI introduced in Grafana 9.3 and why this is important for us to consolidate our security efforts.

### CODE REVIEW

**Review by aarongodin — APPROVED** [review]
The code looks fine to me. It was a little confusing to figure out because of your claim of the breaking change:

> This is a breaking change for users who use uppercase in their login or emails. The users are by default now using lowercase as part of their login and emails.

Am I correct in thinking that it's only a breaking change if for some reason you had two user records with the same characters but different casing? (e.g. one user record for `agodin@grafana.com` and another for `aGodin@grafana.com`).

Before this code change, I guess you'd be able to still log in as either, so it makes sense that it is a breaking change in that regard. Just want to clarify my understanding.

**Comment by eleijonmarck** [comment]
> The code looks fine to me. It was a little confusing to figure out because of your claim of the breaking change:
> 
> > This is a breaking change for users who use uppercase in their login or emails. The users are by default now using lowercase as part of their login and emails.
> 
> Am I correct in thinking that it's only a breaking change if for some reason you had two user records with the same characters but different casing? (e.g. one user record for `agodin@grafana.com` and another for `aGodin@grafana.com`).
> 
> Before this code change, I guess you'd be able to still log in as either, so it makes sense that it is a breaking change in that regard. Just want to clarify my understanding.

thx! i clarified the breaking change a bit more, to account for users reading this in the change log. Thx for the review

File: pkg/services/user/userimpl/store.go [hunk 1]
```diff
@@ -257,10 +242,8 @@ func (ss *sqlStore) GetByEmail(ctx context.Context, query *user.GetUserByEmailQu
...
 		}
 
-		if ss.cfg.CaseInsensitiveLogin {
-			if err := ss.userCaseInsensitiveLoginConflict(ctx, sess, usr.Login, usr.Email); err != nil {
-				return err
-			}
+		if err := ss.userCaseInsensitiveLoginConflict(ctx, sess, usr.Login, usr.Email); err != nil {
+			return err
 		}
 		return nil
```
Comment (linoman) [inline pkg/services/user/userimpl/store.go:248]: Call to function can be inlined

File: pkg/setting/setting.go [hunk 2]
```diff
@@ -1657,7 +1657,9 @@ func readUserSettings(iniFile *ini.File, cfg *Cfg) error {
 			string(roletype.RoleAdmin)})
 	cfg.VerifyEmailEnabled = users.Key("verify_email_enabled").MustBool(false)
 
-	cfg.CaseInsensitiveLogin = users.Key("case_insensitive_login").MustBool(true)
+	// Deprecated
+	// cfg.CaseInsensitiveLogin = users.Key("case_insensitive_login").MustBool(true)
+	cfg.CaseInsensitiveLogin = true
```
Comment (linoman) [inline pkg/setting/setting.go:1662]: This option can just be removed

**Review by linoman — APPROVED** [review]

**Pull request merged**